Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/ReflectometryISISSumBanks.py

```diff
@@ -44,17 +44,12 @@ def PyInit(self):
 
     def PyExec(self):
         input_workspace = self.getProperty(self._WORKSPACE).value
+        num_banks = self._get_rectangular_detector_component(input_workspace).xpixels()
 
-        if not self.getProperty(self._ROI).isDefault:
-            roi_detector_ids = self.getProperty(self._ROI).value
-            masked_workspace = self.mask_detectors(input_workspace, roi_detector_ids)
-            summed_workspace = self.sum_banks(masked_workspace)
+        if num_banks > 1:
+            self.exec_multi_bank(input_workspace)
```

Comment:
Assuming I'm following through correctly, I'm wondering if something like the following (which I haven't tested, sorry!) might help to reduce some of the code duplication that we have between the `exec_multi_bank` and `exec_single_bank` methods:

```
if not self.getProperty(self._ROI).isDefault:
      roi_detector_ids = self.getProperty(self._ROI).value
      masked_workspace = self.mask_detectors(input_workspace, roi_detector_ids)
else:
      masked_workspace = input_workspace

if num_banks > 1:
      summed_workspace = self.sum_banks(masked_workspace)
      result = self._prepend_monitors(input_workspace, summed_workspace)
      self.setProperty(self._OUTPUT_WS, result)
else:
      self.setProperty(self._OUTPUT_WS, masked_workspace)
```